### DISCUSSION TITLE
MustWalk and the animation-less scuba suit cause a walking animation lock.

### DISCUSSION THREAD

**Discussion by HakureiReimu18** [FakeFishGames/Barotrauma#13846] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

 This is a bug related to MustWalk. In the game, I wrote a new Affliction that applies the MustWalk effect using AbilityFlag, which worked fine in previous versions. However, after the recent character skeleton animation update, if this Affliction is repeatedly applied, and then the character wears a scuba suit that does not change the character's animation, and waits for the effect to wear off before applying it again, the character's walking animation will still be the default Vanilla scuba suit walking animation, regardless of whether the scuba suit is worn or not. This is the link to my mod's Steam Workshop page:https://steamcommunity.com/sharedfiles/filedetails/?id=2941124988

### Reproduction steps

1.In my mod, there is an affliction associated with a character's clothing. Wearing it applies several afflictions, one of which restricts the character to walking only, utilizing MustWalk. In my mod, there are two pieces of equipment and a new skill level that, when their conditions are met, can remove this debuff.
2.I opened the submarine editor and prearranged items in it, then started testing. First, I had the character wear the "Patchouli's Outfit+" which grants an additional talent and continuously applies "The_Still_Library" affliction.
3.I entered the command "setskill Touhou_Magic max" in the F3 console to maximize the skill level of the new skill added by your mod. Then, I helped you equip the "Magic_Pouch" and "Magic_Veil" so that you can consistently obtain the "Touhou_Magic_Boost" effect. When this effect reaches its maximum value, your character will become immune to the affliction called "MustWalk," allowing them to regain their running ability.
4.The problem lies here. When the character becomes immune to "MustWalk," their animation is normal. However, if I remove the "Magic_Pouch" and the character loses the immunity to "MustWalk," and then apply the "MustWalk" effect again, the character's walking animation will change to the animation used when wearing a diving suit. This also happens when I remove the "Magic_Veil." Regardless of what actions I take, the walking animation remains the same, and only the walking animation changes in this manner.

https://github.com/FakeFishGames/Barotrauma/assets/140368777/0afad10a-7249-4683-a44d-016119a3bb34

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.4.1 (Blood in the Water Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I gave the mod a test, but unfortunately I couldn't follow the repro steps: there doesn't seem to be an item called "Patchouli's Outfit+" nor an affliction called "MustWalk" in the mod.

**Reply by HakureiReimu18** [reply]
I'm sorry, this is my mistake. Because I wrote the color file, its actual name should be "‖color:120,75,100,255‖Patchouli's Outfit+‖color:end‖".This clothing is located in "Items/Clothes/Patchouli_gear.xml".

In addition, please let me explain the Affliction related to MustWalk: When wearing "‖color:120,75,100,255‖Patchouli's Outfit+‖color:end‖", it will continue to give the Affliction "The Great Unmoving Library", with the id "The_Still_Library", which can be found in line 393 of the "Touhou_Character_Afflictions.xml" file. It will give a new affliction, with the id "The_Still_Library_Walk_NotInWater", located in line 424 of the "Touhou_Character_Afflictions.xml". This Affliction uses "<AbilityFlag flagtype="MustWalk" />", which is the method I use to make the character MustWalk. I hope these additional details can provide enough information to help you find the bug.


`    <Affliction identifier="The_Still_Library" type="Touhou_Character_Affliction" isbuff="true" limbspecific="false" hideiconafterdelay="false" maxstrength="10" iconcolors="126,211,224;126,211,224;126,211,224;126,211,224">
        <Effect minstrength="0" maxstrength="10" resistancefor="damage,bloodloss,bleeding" minresistance="-0.2" maxresistance="-0.2" minbuffmultiplier="2.0" maxbuffmultiplier="2.0"
                strengthchange="-2">
            <StatValue stattype="SwimmingSpeed" value="0.6" />
            <StatValue stattype="WalkingSpeed" value="-0.2" />
            <StatValue stattype="MaximumHealthMultiplier" value="-0.3" />
            <StatValue stattype="SkillGainSpeed" value="0.5" />
            <StatValue stattype="MedicalItemEffectivenessMultiplier" value="0.1" />
            <StatValue stattype="MeleeAttackMultiplier" value="-1" />
            
            <StatusEffect type="NotInWater" target="Character" comparison="And" >
                <Conditional Touhou_MEID_Effect="eq 0"/>
                <Conditional Touhou_Magic_Boost="lt 290"/>
                <Affliction identifier="The_Still_Library_Walk_NotInWater" amount="10"/>
            </StatusEffect>
        </Effect>
        <Icon texture="%ModDir%/Images/Afflictions/The_Still_Library.png" sourcerect="0,0,128,128" color="255,255,255,255" />
    </Affliction>`

`    <Affliction identifier="The_Still_Library_Walk_NotInWater" type="Touhou_Character_Affliction" isbuff="true" limbspecific="false"
                showiconthreshold="100" showinhealthscannerthreshold="100" hideiconafterdelay="false" maxstrength="10" iconcolors="126,211,224;126,211,224;126,211,224;126,211,224">
        <Effect minstrength="0" maxstrength="10" strengthchange="-4">
            
            <AbilityFlag flagtype="MustWalk" />
        </Effect>
        <Icon texture="%ModDir%/Images/Afflictions/The_Still_Library.png" sourcerect="0,0,128,128" color="255,255,255,255" />
    </Affliction>`

**Comment by Regalis11** [comment]
Thank you for the extra info! I created [a ticket about this in the issue tracker](https://github.com/FakeFishGames/Barotrauma/issues/13901), we'll look into it!

**Comment by Regalis11** [comment]
I finally managed to take a look at this, and after doing some testing, I believe there isn't actually a bug here: this isn't the diving suit animation, it's the "vigorous walk" caused by the hyperactivity affliction which I believe the character gets from the high level of magic boost.
